plr edits (#70)
popsim-consortium/adding-species-manuscript: adding_species_manuscript.tex

The initial version of \stdpopsim focused on establishing this framework using 
six well-characterized model species.
Here, we report on major updates made in the new release of \stdpopsim (version 0.2),

> which includes a significant expansion of the species catalog and substantial additions to simulation capabilities.

Author: it's not yet clear why adding gene conversion makes it possible to simulate more species, so this was confusing, i thought

Reviewer: This is mostly a reference to bacterial recombination. Indeed, not all simulation capabilities that were added have to do with expanding the set of species. I'm flagging @lauterbur here, because I think it was text that she suggested.

Author: I understood it, and it's a good thought, but I didn't think it'd make sense to the reader at this point.

Author: (happy to stick it back in though if someone thinks we should)

Reviewer 2: Since we decided we're keeping the discussion of new features to those that are specific to the species catalog, my thought was to add something to that effect. (I think we say elsewhere that while there are also a bunch of additions for selection, we're explicitly not discussing those here, for instance.) I see your point that it isn't immediately clear with just this added clause. I don't know that it's confusing though, and is clarified somewhat in the next sentence.

Reviewer: The section in the paper does a good  in tying the added features to the expanded catalog. We don't necessarily have to explicitly make this point in the abstract. The problem with the statement that @petrelharp removed was that it implied that all (or most) of the features we added serve the purpose of allowing simulation of a wider range of species. In fact, this strictly true only for bacterial recombination. We also have finite site mutation models and gene conversion, which do not strictly allow modeling other species. So, I vote to keep @petrelharp's version here.

which includes a significant expansion of the species catalog and substantial additions to simulation capabilities.
Features added to improve the realism of the simulated genomes include non-crossover recombination and provision of species-specific genomic annotations.
Through community-driven efforts, we expanded the number of species in the catalog by more than three-fold and broadened coverage across the tree of life.
Our experience throughout this process was that people are keen to put in the time and effort to include their study species,
but that simple, clear guidance is vital.
Consequently, our intention with this paper is also to provide a learning

> modality to meet that need,

Author: A "learning modality" is a *way* of learning things, whereas we have a set of guidelines. Also, the first sentence here seemed a bit condescending, so I rephrased.

Reviewer: I agree with changing "modality" to "guidelines", but I like preceding this with a statement that ties these guidelines to a need that came up during the expansion of the catalog. This gets a bit lost in the shorter version. I couldn't think of a good compromise. @lauterbur? Do you have a suggestion here?

Author: See suggestion below.

> In this paper, we aim to provide guidance for others aiming to do genome-scale simulations
> by reporting on the main lessons learned through this process.

Author: ```suggestion
During the process of expanding the catalog,
we have identified common sticking points and developed best practices for setting up genome-scale simulations. 
```

Reviewer 2: This works for me, so does the previous version.

Reviewer: "wide variety of other researchers and organisms"? Do we not consider all researchers to be organisms? ;-)
I'm not sure we need to emphasize the variety of researchers or species added here. Also, many of the "other researchers" are actually part of the author list, so we may not wish to "other" them. It might just be enough to say
> During the process of expanding the catalog, we have identified . . .  

The short sentence after that (In this paper, ...) becomes a bit redundant now. I think that the following sentence ("We describe ...") connects quite well this the previous one ("... setting up genome-wide simulations."). What do you say?

Author: Good call; I've updated my suggestion above accordingly.

more and more detailed elements of a species' biology.
Important elements include genomic features such as mutation and recombination
rates that strongly affect genetic variation and haplotype structure
\citep{Nachman2002}, which have particularly strong ramifications 
when linked selection is important in the patterns of genomic diversity being studied \citep{Cutter2013}.
Furthermore, the demographic history of a species,
encompassing population sizes and distributions, divergences, and gene flow, can
dramatically affect patterns of genomic variation \citep{Teshima2006}. Thus
species-specific estimates of these and other ecological and evolutionary parameters 
(e.g., those governing the process of natural selection) 

> are important to realistic simulations.

Author: I don't think they're quite "fundamental" (i.e., everything else is built on them).

Reviewer: Agree, but "important to realistic simulations" reads a bit funny to me. How about "important when generating realistic simulations"?

Reviewer 2: I do think they're fundamental for "realistic" simulations, but I'm fine with changing the language here.

> are important to realistic simulations.

Author: ```suggestion
are important when generating realistic simulations.
```

and if necessary with input of additional members of the community.
This quality control process quite often finds subtle bugs \citep[e.g., as in][]{Ragsdale2020}
or highlights parts of the model that are ambiguously defined by the literature sources.

> This increases the reproducibility of the resulting simulations in any downstream analysis.

Author: ```suggestion
This increases the reliability and reproducibility of the resulting simulations in any downstream analysis.
```

> 

Reviewer: I think both versions are fine, but you get reproducible simulations even if the code did not undergo any QC. IMO, the QC is mostly meant to improve the reliability of the simulations in the sense that the user can rely on the fact that they provide an honest implementation of the information in the cited sources.

Author: Ah, okay. I was thinking about "reliable" as meaning that they'd, like, reliably run and not crash or something. Let's do "reliability and reproducibility". (since it is more reproducible if you know exactly what they're doing)

\citep[e.g.,][]{McVicker2009,Charlesworth2012}. 
In addition, the lengths of chromosome-scale shared haplotypes within and
between populations provides valuable information on their demographic history.
Demographic inference methods that use such information,
such as MSMC \citep{Schiffels2020}, and IBDNe \citep{browning2015accurate},
perform best on long genomic segments with realistic recombination rates.
Chromosome-scale simulations are clearly required to test (or train) such methods,
or to conduct power analyses when designing empirical studies that use them.

\Stdpopsim makes these simulations available
through only a single call to a command-line script
or to a handful of lines of python code,

> along with citations to all primary sources underlying the simulation.

Author: It felt like we needed a conclusion sentence here.

many of whom expressed interest in adding non-model species to \stdpopsim.
The hackathon was then structured based on feedback from these participants.
One month before the hackathon, we organized a final workshop to prepare interested
participants, by introducing them to  the process of developing
a new species model and adding it to the \stdpopsim code base.

> Roughly 20 scientists participated in the hackathon (and are included as authors on this paper),

Author: I think they all are?

Reviewer: That's one for @lauterbur

Reviewer 2: We have a few who declined authorship.

Reviewer: I think that we can keep this statement as is even if not all participants agreed to be authors. I assume that everyone who ended contributing new species or demo models were included as authors.

> Roughly 20 scientists participated in the hackathon (and are included as authors on this paper),

Author: ```suggestion
Roughly 20 scientists participated in the hackathon (most of whom are included as authors on this paper),
```

  At minimum, an average genome-wide recombination rate needs to be specified, which is typically available for well assembled genomes.
  For bacteria and archea, which primarily experience non-crossover recombination,
  the average tract length should also be specified

>   (see details in previous section).

Author: The indentation was weird with a linebreak, I thought - inline in the paragraph is okay.

Therefore, its incorporation into simulations is crucial for many purposes.
To achieve this, the simulator needs to know which regions along the genome are subject to selection,
and the nature and strength of this selection.
The development version of \stdpopsim enables simulation with selection
(using the \texttt{SLiM} engine)
by specifying genome annotations and distributions of fitness effects,
as specified below.
The ability to simulate chromosomes with realistic models of
selection is still under development and will be finalized in the next release of \stdpopsim,

> and will be described in more detail in a future paper.

Reviewer: you have two subsequent 'and's here. Should we switch the second one to a 'which' ("which will be described...")?

> The ability to simulate chromosomes with realistic models of
> selection is still under development and will be finalized in the next release of \stdpopsim,
> and will be described in more detail in a future paper.

Author: ```suggestion
The ability to simulate chromosomes with realistic models of
selection is still under development. It will be finalized in the next release of \stdpopsim,
and described in more detail in a future paper.
```

%Ilan: tried to give a few concrete examples here
If, for example, the main purpose of the simulation is to examine the distribution of lengths of shared haplotypes between individuals,
or study patterns of background selection,
then it makes sense to simulate such pseudo chromosomes.
Note, however, that specific genetic correlations between different contigs lumped together can be highly inaccurate.
So, if the main purpose of the simulation is to examine local patterns of genetic variation in loci of interest, then it would be more appropriate to simulate the relevant contigs separately (even if they are short), or to randomly sample several mappings of contigs to pseudo-chromosomes.

> Finally, we note that for some purposes it may be sufficient to simulate a large number of unlinked sites \citep{Gutenkunst2009,Excoffier2013},

Author: Perhaps this is true, but tangential - I think to say this we should make clear what "some purposes" are, which is beyond the scope of this bit.

Reviewer: The main reason to add this statement is to give people a last-resort option if they have popgen data but no genome assembly, like in RAD-SEQ studies. They might still be able to infer demographic histories and simulate data. I'm fine with leaving this out, so that this paragraph is focused on the more interesting cases of partial builds.

Author: Good call, but I don't think we're saying people *can't* do single loci? We're not trying to review all possible simulation methods? (Or, if we do, we need some more context?)

Reviewer: Let's keep this commented out for now, and see if we miss this in future editing rounds. If needed, we can add this with some more detail, like saying that tasks such as demographic inference and DFE inference can be done using unlinked sites.

Reviewer 2: I think the sentence is both useful and tangential - this paper isn't just about adding species to stdpopsim, it's also promoting standardized, species-specific simulations in general, which will appeal to anyone doing empirical research, including people with eg. RADSeq data. But those people aren't who the paper is for, and I think it's valid to make that pretty explicit. Perhaps "However, this approach would not have the many benefits of whole-chromosome simulations, which we promote in \stdpposim. While some of the same considerations hold, a discussion of such simulations goes beyond the scope of this paper."

Reviewer: I think that @petrelharp is raising another issue - that we should not suggest that simulation of independent sites is useful without explicitly saying when it is possibly useful. It's not like people will stop doing independent site simulations, and there are probably a few things that such simulations can benefit from by using our guidelines (QC, fitting a demographic model, mutation rate), but we don't necessarily need to go out of our way to make this point explicitly. My suggestion is to leave this part of the text commented out and return to it in a clean read. If it still looks to us like something's missing here, then we can figure a way to make this point the "right" way.

Author: See suggestion below.

> % Finally, we note that for some purposes it may be sufficient to simulate a large number of unlinked sites \citep{Gutenkunst2009,Excoffier2013},
> % which can be generated without any sort of genome assembly.
> % However, this approach would not have the many benefits of whole-chromosome simulations,
> % which we promote in \stdpopsim.

Author: ```suggestion
For some purposes it makes sense to simulate a large number of unlinked sites \citep{Gutenkunst2009,Excoffier2013},
which can be generated without any sort of genome assembly.
However, this approach would not have the benefits of whole-chromosome simulations.
While some of the same considerations hold, a discussion of the advantages and disadvantages of such simulations goes beyond the scope of this paper.
```

\citet{zheng1996integrated} (Figure \ref{fig:anogam}A).
As direct estimates of \textbf{mutation rate} (e.g., via mutation accumulation) do not currently exist for \emph{Anopheles gambiae},
we used the genome-wide average mutation rate of $\mu=3.5 \times 10^{-9}$ mutations per generation per site,

> estimated by \cite{Keightley2009} for the fellow Dipteran \textit{Drosophila melanogaster},

Author: mosquito-fly is what, like 40Mya?

Reviewer: 'closely related' is always very vague. We added this here just to signal that we chose D. mel because of its close relation to A. gam. "fellow Dipteran" is a better choice.

accompanied by the relevant citation information,
and the model underwent the standard quality control process.
%Ilan: any issues worth mentioning that were raised in the QC?

> The species may be refined in the future by adding more demographic models,

Reviewer: I don't assume that anyone is going to think we're considering modifying the actual mosquito species, but "refining a species" sounds a bit odd ;-).

Author: I changed it because "model" appeared too many times - how about "species entry"?

Reviewer: Species entry sounds good

> The species may be refined in the future by adding more demographic models,

Author: ```suggestion
The species entry may be refined in the future by adding more demographic models,
```

We also discuss special considerations for collecting relevant information from the literature,
and what to do if some of that information is not available.
Because this process is quite error-prone,
we encourage wider adoption of ``code review'',
i.e., for researchers implementing simulations to have their
parameter choices and implementation reviewed by at least one other researcher.

> %% PLR: It's not totally clear which "these steps" are, and they are actually required for stdpopsim?

Reviewer: "These steps" refers to our guidelines, and not just the QC part. This statement is important, but could be clarified by changing "These steps" to "The guidelines we provide in this paper". Some of these guidelines are strict requirements of stdpopsim and some are not. I think it's okay not to make this distinction here.

Author: Ah, I see. I was shying a little away from saying "everyone please add all your simulation code to stdpopsim" since it can be hard to find QC'ers. But, how's this?

Reviewer: Looks good. I do think we should be encouraging people to add their code to stdpopsim. If they read this paper, then they know what our requirements and standards are, so I don't think we are risking open the door too wide here.

> % These steps can be followed if you are implementing a simulation independently for your own use,
> % or, as we encourage, when you add your simulation code to \stdpopsim,
> % to ensure its robustness and make it available for future standardized research.

Author: ```suggestion
The guidelines in this paper can be followed when implementing a simulation independently for a single study,
or (as we encourage others to do) when adding code to \stdpopsim,
which helps to ensure its robustness and to make it available for future research.
```
